multichain-testing/scripts/fast-usdc.ts

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env -S node --import ts-blank-space/register
+import '@endo/init';
+import { parseArgs } from 'node:util';
+import type { ExecutionContext } from 'ava';
+import { commonSetup } from '../test/support.js';
+import { makeFeedPolicy, oracleMnemonics } from '../test/fast-usdc/config.js';
+import { makeDenomTools } from '../tools/asset-info.js';
+import { makeDoOffer } from '../tools/e2e-tools.js';
+import { AmountMath, type Brand } from '@agoric/ertp';
+import { divideBy } from '@agoric/zoe/src/contractSupport/ratio.js';
+import type { USDCProposalShapes } from '@agoric/fast-usdc/src/pool-share-math.js';
+import type { PoolMetrics } from '@agoric/fast-usdc/src/types.js';
+
+const contractName = 'fastUsdc';
+const contractBuilder =
+  '../packages/builders/scripts/fast-usdc/init-fast-usdc.js';
+
+const mockT = {
+  log: console.log,
+  is: (condition: unknown, expected: unknown, message: string) => {
+    if (condition !== expected) {
+      throw new Error(
+        `Condition: ${message} failed. Expected ${expected} got ${condition}.`,
+      );
+    }
+  },
+} as unknown as ExecutionContext;
+
+// from ../test/fast-usdc/fast-usdc.test.ts
+type VStorageClient = Awaited<ReturnType<typeof commonSetup>>['vstorageClient'];
+const agoricNamesQ = (vsc: VStorageClient) =>
+  harden({
+    brands: <K extends AssetKind>(_assetKind: K) =>
+      vsc
+        .queryData('published.agoricNames.brand')
+        .then(pairs => Object.fromEntries(pairs) as Record<string, Brand<K>>),
+  });
+
+const fastLPQ = (vsc: VStorageClient) =>
+  harden({
+    metrics: () =>
+      vsc.queryData(`published.fastUsdc.poolMetrics`) as Promise<PoolMetrics>,
+    info: () =>
+      vsc.queryData(`published.${contractName}`) as Promise<{
+        poolAccount: string;
+        settlementAccount: string;
+      }>,
+  });
+
+const parseCommandLine = () => {
+  const { values, positionals } = parseArgs({
+    options: {
+      oracle: {
+        type: 'string',
+      },
+      mnemonic: {
+        type: 'string',
+      },
+      help: {
+        type: 'boolean',
+      },
+    },
+    allowPositionals: true,
+  });
+
+  if (values.help) {
+    console.log(`
+Usage:
+  No arguments      - start the contract and fund the liquidity pool
+  start             - only start the contract
+  fund-pool         - fund the FUSDC Liquidity Pool
+  provision-wallet  - provision a smart wallet (requires --mnemonic)
+  fund-faucet       - fund the faucet account with bridged USDC
+  --oracle          - Comma-separated list of oracle addresses
+
+Examples:
+  ./fast-usdc.ts
+  ./fast-usdc.ts --oracle oracle1:addr1,oracle2:addr2
+  ./fast-usdc.ts start
+  ./fast-usdc.ts start --oracle oracle1:addr1,oracle2:addr2
+  ./fast-usdc.ts fund-pool
+  ./fast-usdc.ts fund-faucet
+`);
+    process.exit(0);
```

Comment:
I'd rather we don't hide use of this very powerful `process.exit` authority under the name `parseCommandLine()`. Return `undefined` or throw or something instead, please?